src/code/z_room.c

```diff
@@ -317,33 +347,37 @@ void Room_Draw1Single(PlayState* play, Room* room, u32 flags) {
 
     activeCam = GET_ACTIVE_CAM(play);
     isFixedCamera = (activeCam->setting == CAM_SET_PREREND_FIXED);
-    meshHeader1Single = &room->meshHeader->meshHeader1Single;
-    meshHeader1Entry = SEGMENTED_TO_VIRTUAL(meshHeader1Single->base.entry);
-    drawBg = (flags & ROOM_DRAW_OPA) && isFixedCamera && meshHeader1Single->source && !(SREG(25) & 1);
-    drawOpa = (flags & ROOM_DRAW_OPA) && (meshHeader1Entry->opa != NULL) && !(SREG(25) & 2);
-    drawXlu = (flags & ROOM_DRAW_XLU) && (meshHeader1Entry->xlu != NULL) && !(SREG(25) & 4);
+    meshHeaderPrerenderSingle = &room->meshHeader->prerenderSingle;
+    dListsEntry = SEGMENTED_TO_VIRTUAL(meshHeaderPrerenderSingle->base.entry);
+    drawBg = (flags & ROOM_DRAW_OPA) && isFixedCamera && (meshHeaderPrerenderSingle->source != NULL) &&
+             !(R_ROOM_PREREND_NODRAW_FLAGS & (1 << 0));
+    drawOpa = (flags & ROOM_DRAW_OPA) && (dListsEntry->opa != NULL) && !(R_ROOM_PREREND_NODRAW_FLAGS & (1 << 1));
+    drawXlu = (flags & ROOM_DRAW_XLU) && (dListsEntry->xlu != NULL) && !(R_ROOM_PREREND_NODRAW_FLAGS & (1 << 2));
```

Comment:
No defines for these flags?

Reply:
I don't feel like it's worth introducing flags and enums for debug features? it's usually simple enough that it's self-documenting like here

If you insist though, sure. Where to put them?

Reply:
The natural place seems to be with the rest of the room draw stuff in the scene header. I think it is worth having stuff for debug features since those are often one of the first things people want to mess with (after they give up on Player, at least).